addon/components/inputs/datetime.js

```diff
@@ -57,8 +69,47 @@ export default AbstractInput.extend({
     return moment(value).format(DATE_TIME_FORMAT)
   },
 
+  // == Lifecycle Hooks =======================================================
+
+  init () {
+    this._super(...arguments)
+
+    const value = this.get('value')
+    const timezone = this.get('timezone')
+    let currentDateTime = getMomentTimezone(value, timezone)
+
+    if (!this.get('defaultToCurrentDateTime')) {
+      currentDateTime = value
+    } else {
+      currentDateTime = getFormattedTime(currentDateTime, this.get('dateTimeFormat'), timezone)
+    }
+
+    this.setProperties({
+      localTimezone: moment().format('Z')
+    })
+
+    if (!isNone(value)) {
```

Comment:
You should only call `onChange` if the value is different previously. For example, if the initial value was `null` or `undefined`, you'd want to call `onChange`.  If `value` is `none`, then you'd want to call `onChange`.

Reply:
Yeah, that makes sense. I think that this and the stuff you commented on above actually aren't needed. Not quite sure what I was thinking behind this

Reply:
That's quite alright. I think the idea was sane. Just needed a little more polish to make sure it worked properly. I would have been okay if you left it but corrected it. As it stands, the component does suffer from not setting the value when the user wants defaults applied. The `currentValue` is only presentational until the first date selection is made.

Leaving it out is fine as well since it's only returning to the same behavior as before.